### PR TITLE
feat!: now can specify whether we want curve checks during zkey deser

### PR DESCRIPTION
**Pull request by 0xThemis**
BREAKING CHANGE: changed the traits for circom bridge. Also modified the from_reader impl for the two Zkeys

### CODE REVIEW

File: co-circom/circom-types/src/traits.rs [hunk 1]
```diff
@@ -508,7 +524,7 @@ where
     phantom_data: PhantomData<P>,
 }
 
-impl<P: Pairing + CircomArkworksPairingBridge> G2Visitor<P>
+impl<P: Pairing + CircomArkworksPairingBridge> TargetGroupVisitor<P>
```
Comment (dkales) [inline co-circom/circom-types/src/traits.rs:527]: what is this for?

File: co-circom/co-circom/src/bin/co-circom.rs [hunk 2]
```diff
@@ -477,7 +477,10 @@ where
 
     let public_input = match proof_system {
         ProofSystem::Groth16 => {
-            let zkey = Arc::new(Groth16ZKey::<P>::from_reader(zkey_file).context("reading zkey")?);
+            let zkey = Arc::new(
+                Groth16ZKey::<P>::from_reader(zkey_file, CheckElement::Yes)
```
Comment (dkales) [inline co-circom/co-circom/src/bin/co-circom.rs:481]: this should probably be configurable from env

**Review by dkales — APPROVED** [review]

File: co-circom/circom-types/src/traits.rs (hunk 1 above)
Comment (0xThemis) [reply co-circom/circom-types/src/traits.rs:527]: just moving stuff around

File: co-circom/co-circom/src/bin/co-circom.rs (hunk 2 above)
Comment (0xThemis) [reply co-circom/co-circom/src/bin/co-circom.rs:481]: we agreed to move it to later

**Pull request merged**